admin-frontend/open_admin_app/lib/widgets/feature-groups/feature-group-settings.dart

```diff
@@ -0,0 +1,311 @@
+                    return FeaturesDropDown(
+                        features: snapshot.data!, bloc: bloc);
+                  } else {
+                    return const SizedBox.shrink();
+                  }
+                }),
+            SizedBox(
+              width: 8.0,
+            ),
+            TextButton.icon(
+                icon: const Icon(Icons.add),
+                label: const Text('Add Feature'),
+                onPressed: () => {_addFeatureToGroup(bloc)}),
+          ],
+        ),
+        StreamBuilder<List<FeatureGroupFeature>>(
+            stream: bloc.groupFeaturesStream,
+            builder: (context, snapshot) {
+              if (snapshot.hasData && snapshot.data!.isNotEmpty) {
+                return Column(
+                  children: [
+                    for (FeatureGroupFeature feature in snapshot.data!)
+                      Padding(
+                        padding: const EdgeInsets.symmetric(
+                            horizontal: 8.0, vertical: 16.0),
+                        child: Card(
+                          child: Padding(
+                            padding: const EdgeInsets.all(8.0),
+                            child: Container(
+                              height: 42,
+                              child: Row(
+                                children: [
+                                  Text(feature.name),
+                                  SizedBox(width: 8.0),
+                                  FeatureValueContainer(
+                                      bloc: bloc, feature: feature),
+                                ],
+                              ),
+                            ),
+                          ),
+                        ),
+                      ),
+                  ],
+                );
+              } else {
+                return const SizedBox.shrink();
+              }
+            })
+      ],
+    );
+  }
+
+  _addFeatureToGroup(FeatureGroupBloc bloc) {
+    bloc.addFeatureToGroup();
+  }
+}
+
+class FeatureValueContainer extends StatelessWidget {
+  final FeatureGroupFeature feature;
+  final FeatureGroupBloc bloc;
+
+  const FeatureValueContainer(
+      {Key? key, required this.bloc, required this.feature})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    switch (feature.type) {
+      case FeatureValueType.STRING:
+        return EditFeatureGroupStringValueContainer(
+          editable: true,
+          feature: feature,
+          bloc: bloc,
+        );
+      case FeatureValueType.BOOLEAN:
+        return EditFeatureGroupBooleanValueWidget(
+          editable: true,
+          feature: feature,
+          bloc: bloc,
+        );
+      case FeatureValueType.NUMBER:
+        // TODO: Handle this case.
+        break;
+      case FeatureValueType.JSON:
+        // TODO: Handle this case.
+        break;
+      default:
+        "foo";
```

Comment:
baaaaaaaaahhhhhhhhhh 🐑

Reply:
I expect it asks you to do this because the feature.type could be null? If it can't be null, adding the ! will get rid of the "default" section I believe.